app/controllers/directory_columns_controller.rb

```diff
@@ -1,15 +1,20 @@
 # frozen_string_literal: true
 
 class DirectoryColumnsController < ApplicationController
-  requires_login
+  requires_login except: [:index]
 
   def index
+    directory_columns = DirectoryColumn.includes(:user_field).where(enabled: true).order(:position)
+    render_json_dump(directory_columns: serialize_data(directory_columns, DirectoryColumnSerializer))
+  end
+
+  def edit_index
```

Comment:
Kinda weird method name here... but we have 2 indexes. One is public index and one is edit index so I went with it.

Reply:
This is a sign that you should create a new controller. It maps to your URLs below too. You could have `DirectoryColumnsController` and `EditDirectoryColumnsController`.